### PR TITLE
New CI with optimized docker build

### PR DESCRIPTION
**Pull request by plexoos**

### CODE REVIEW

**Comment by plexoos** [comment]
Tried to merge the current main into this but found that it is much easier to deal with the conflicts by rebasing instead

File: docker/Dockerfile.root6 [hunk 1]
```diff
@@ -40,7 +34,11 @@ RUN source star-spack/setup.sh \
  && spack env create star-env-root6 star-spack/environments/star-x86_64-root-6.16.00-container.yaml \
  && spack env activate star-env-root6 \
  && spack install -j 5 --fail-fast \
- && spack gc -y
+ && spack gc -y \
+ && spack clean --all \
+ && spack uninstall --force -y llvm \
+ # Create star-spack/spack/var/spack/environments/star-env-root5/loads
+ && spack -e star-env-root5 env loads
```
Comment (veprbl) [inline docker/Dockerfile.root6]: star-env-root6?

Comment (veprbl) [reply docker/Dockerfile.root6]: NVM, it's renamed in the next commit

Comment (plexoos) [reply docker/Dockerfile.root6]: Github marked this diff as Outdated. Somehow, you are looking at an old commit... Yes, force pushing is evil, confusing, and should be avoided

**Review by veprbl — APPROVED** [review]
It's a big changeset, but only affects CI. Diff LGTM.

File: docker/Dockerfile.root6 (hunk 1 above)
Comment (plexoos) [reply docker/Dockerfile.root6]: > NVM, it's renamed in the next commit

You are actually correct. Sorry for the noise

**Pull request merged**